tests/unit/test_sql_execution.py

```
@@ -1019,3 +1021,241 @@ def test_databricks_connector_dialect_alias_is_registered(self):
+        self.assertEqual(mock_retry_sleep.call_count, 1)
+
+    @mock.patch("urllib3.util.retry.Retry.sleep", return_value=None)
+    @mock.patch("urllib3.connectionpool.HTTPConnectionPool._make_request")
+    @mock.patch("deepnote_toolkit.sql.sql_execution.get_project_auth_headers")
+    @mock.patch("deepnote_toolkit.sql.sql_execution.get_absolute_userpod_api_url")
+    def test_federated_auth_exhausts_retries_on_persistent_5xx(
+        self,
+        mock_get_url,
+        mock_get_headers,
+        mock_make_request,
+        mock_retry_sleep,
```

Ruff (0.15.9)

[warning] Unused method argument: `mock_retry_sleep`

(ARG002)